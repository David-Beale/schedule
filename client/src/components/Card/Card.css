.card-container{
  box-sizing: border-box;
  height:350px;
  width:350px;
  /* margin: 25px; */
  border:1px solid rgba(185, 184, 184, 0.308);
  background-color: rgb(255, 255, 255);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  color: rgb(63, 63, 63);
  transition: 0.6s;
}
.card-container:hover{
  /* height:370px;
  width:370px;
  margin: 15px; */
  box-shadow: 1px 1px 8px 1px rgba(189, 189, 189, 0.75);
}
.card-image{
  width: 100%;
  /* max-height: 50%; */
}
.image-container{
  overflow: hidden;
  height:50%;
}
.card-time{
  font-family: 'Palanquin', sans-serif;
  color: #f05537;
  font-size: 1rem;;
  font-weight: 700;
  margin: 1rem 0;
}
.card-title{
  font-family: 'Palanquin', sans-serif;
  color: #000000;
  font-size: 1.3rem;;
  font-weight: 700;
  margin-left: 2rem;
  align-self: flex-start;
}
.flip-card{
  perspective: 1000px;
  height:350px;
  width:350px;
  margin: 25px;
}
.flip-card-inner {
  position: relative;
  width: 100%;
  height: 100%;
  transition: transform 1s;
  transform-style: preserve-3d;
}
.flip {
  transform: rotateY(180deg);
}
.flip-card-front, .flip-card-back {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}
.flip-card-back {
  transform: rotateY(180deg);
}
.back {
  align-self: center;
}
.description-container{
  padding:5%;
  background: linear-gradient(to bottom, #f7f7f7 0%,#f4f8fa 70%,#f2f8fc 100%); 
  width: 90%;
  margin:0 10%;
  overflow: hidden;
  height:40%;
}
.countdown-container{
  width:100%
}
.event-link{
  font-size: 1.5rem;
  position: absolute;
  bottom:1rem;
}