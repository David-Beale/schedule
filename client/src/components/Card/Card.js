import React, { useState } from 'react';
import Moment from 'react-moment';
import Countdown from '../Countdown'
import EventModal from '../EventModal/EventModal';

function Card(props) {
  const [flipped, setFlipped] = useState(false)
  const [modalVisible, setModalVisible] = useState(false);
  const handleClick = () => {
    setFlipped(!flipped)
  }
  const toggleModal = (e) => {
    e.stopPropagation();
    setModalVisible(!modalVisible);
  }
  
  return (
    <div>
      <div className='flip-card'>
        <div onClick={handleClick} className={flipped ? 'flip flip-card-inner' : 'flip-card-inner'} >
          <div className='card-container flip-card-front'>
            <div className='image-container'>
              <img src={props.info.image} className='card-image' />
            </div>
            <div className='card-time'><Moment format="ddd Do MMM, HH:mm">{props.info.date}</Moment></div>
            <div className='card-title'>
              {props.info.title}
            </div>
          </div>

          <div className='card-container flip-card-back'>
            <div className='card-title back'>
              {props.info.title}
            </div>
            <div className='description-container'>
              {props.info.description}
            </div>
            <div className='countdown-container'>
              {flipped &&
                <Countdown
                  info={props.info}
                />
              }
            </div>
            <a onClick={toggleModal} className='event-link'>Click here</a>
          </div>
        </div>

      </div>
      modalVisible ? 
      <EventModal visible={modalVisible} toggleModal={toggleModal} info={props.info} /> : null;
    </div>
  );
}

export { Card as default };
