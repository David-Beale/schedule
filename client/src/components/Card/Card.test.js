import Card from './Card';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import React from 'react';

describe('Card', () => {
  it('should render without crashing', () => {
    const RealDate = Date.now;

    beforeAll(() => {
      global.Date.now = jest.fn(() => new Date('2020-04-26T09:46:15.302Z'));
    });

    afterAll(() => {
      global.Date.now = RealDate;
    });
    jest.mock('moment', () => () => ({
      format: () => '2020-04-26T09:46:15.302Z',
    }));
    const div = document.createElement('div');
    ReactDOM.render(
      <Card
        info={{
          image: 'https://img.src',
          artistName: 'joseph',
          date: '2020-04-26T09:46:15.302Z',
        }}
        key={0}
      />,
      div
    );
  });
  it('should render properly', () => {
    const component = render(
      <Card
        info={{
          image: 'https://img.src',
          artistName: 'joseph',
          date: '2020-04-26T09:46:15.302Z',
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('render without data', () => {
    const component = render(
      <Card
        info={{
          image: undefined,
          artistName: undefined,
          date: '2020-04-26T09:46:15.302Z',
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
