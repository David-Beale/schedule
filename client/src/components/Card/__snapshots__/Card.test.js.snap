// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flip-card"
        >
          <div
            class="flip-card-inner"
          >
            <div
              class="card-container flip-card-front"
            >
              <div
                class="image-container"
              >
                <img
                  class="card-image"
                />
              </div>
              <div
                class="card-artist"
              />
              <div
                class="card-time"
              >
                <time
                  datetime="1587894375302"
                >
                  Sun 26th Apr, 10:46
                </time>
              </div>
              <div
                class="card-title"
              />
              <div
                class="calendar-wrapper"
              >
                <div
                  class="react-add-to-calendar"
                >
                  <div
                    class="react-add-to-calendar__wrapper"
                  >
                    <a
                      class="react-add-to-calendar__button"
                    >
                      <span>
                        Add to Calendar 
                        <i
                          class="react-add-to-calendar__icon--right fa fa-caret-down"
                        />
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card-container flip-card-back"
            >
              <div
                class="card-title back"
              />
              <div
                class="description-container"
              />
              <div
                class="countdown-container"
              />
              <a
                class="event-link"
              >
                Watch here!
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flip-card"
      >
        <div
          class="flip-card-inner"
        >
          <div
            class="card-container flip-card-front"
          >
            <div
              class="image-container"
            >
              <img
                class="card-image"
              />
            </div>
            <div
              class="card-artist"
            />
            <div
              class="card-time"
            >
              <time
                datetime="1587894375302"
              >
                Sun 26th Apr, 10:46
              </time>
            </div>
            <div
              class="card-title"
            />
            <div
              class="calendar-wrapper"
            >
              <div
                class="react-add-to-calendar"
              >
                <div
                  class="react-add-to-calendar__wrapper"
                >
                  <a
                    class="react-add-to-calendar__button"
                  >
                    <span>
                      Add to Calendar 
                      <i
                        class="react-add-to-calendar__icon--right fa fa-caret-down"
                      />
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-container flip-card-back"
          >
            <div
              class="card-title back"
            />
            <div
              class="description-container"
            />
            <div
              class="countdown-container"
            />
            <a
              class="event-link"
            >
              Watch here!
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flip-card"
        >
          <div
            class="flip-card-inner"
          >
            <div
              class="card-container flip-card-front"
            >
              <div
                class="image-container"
              >
                <img
                  class="card-image"
                  src="https://img.src"
                />
              </div>
              <div
                class="card-artist"
              >
                joseph
              </div>
              <div
                class="card-time"
              >
                <time
                  datetime="1587894375302"
                >
                  Sun 26th Apr, 10:46
                </time>
              </div>
              <div
                class="card-title"
              />
              <div
                class="calendar-wrapper"
              >
                <div
                  class="react-add-to-calendar"
                >
                  <div
                    class="react-add-to-calendar__wrapper"
                  >
                    <a
                      class="react-add-to-calendar__button"
                    >
                      <span>
                        Add to Calendar 
                        <i
                          class="react-add-to-calendar__icon--right fa fa-caret-down"
                        />
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card-container flip-card-back"
            >
              <div
                class="card-title back"
              />
              <div
                class="description-container"
              />
              <div
                class="countdown-container"
              />
              <a
                class="event-link"
              >
                Watch here!
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flip-card"
      >
        <div
          class="flip-card-inner"
        >
          <div
            class="card-container flip-card-front"
          >
            <div
              class="image-container"
            >
              <img
                class="card-image"
                src="https://img.src"
              />
            </div>
            <div
              class="card-artist"
            >
              joseph
            </div>
            <div
              class="card-time"
            >
              <time
                datetime="1587894375302"
              >
                Sun 26th Apr, 10:46
              </time>
            </div>
            <div
              class="card-title"
            />
            <div
              class="calendar-wrapper"
            >
              <div
                class="react-add-to-calendar"
              >
                <div
                  class="react-add-to-calendar__wrapper"
                >
                  <a
                    class="react-add-to-calendar__button"
                  >
                    <span>
                      Add to Calendar 
                      <i
                        class="react-add-to-calendar__icon--right fa fa-caret-down"
                      />
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-container flip-card-back"
          >
            <div
              class="card-title back"
            />
            <div
              class="description-container"
            />
            <div
              class="countdown-container"
            />
            <a
              class="event-link"
            >
              Watch here!
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
